crates/db/src/lib.rs

```diff
@@ -15,14 +15,12 @@ mod prost_serde;
 
 pub type DBResult<T> = Result<T, RepositoryError>;
 
-#[derive(Debug, Snafu)]
+#[derive(Debug, Error)]
 pub enum RepositoryError {
-    KeyValueStorage {
-        #[snafu(source)]
-        source: Box<dyn std::error::Error>,
-    },
-    #[snafu(display("Invalid data: [{}]", source))]
-    InvalidData { source: Box<dyn std::error::Error> },
+    #[error("Storage error: {0}")]
+    Storage(#[from] Box<dyn std::error::Error>),
+    #[error("Serialization error: {0}")]
+    Serialization(#[from] serde_json::Error),
```

Comment:
I think we shouldn't use a concrete type of serialization/deserialization error here, as external implementors could use different serialization crates. Also in our case `serde_json` is temporary

Reply:
Renamed this one to `InvalidData`, though `#[from] serde_json::Error` part left untouched

Reply:
@NikitaMasych this one is still not resolved

Reply:
The problem was with `#[from] serde_json::Error` too, again, read the comment above

Reply:
Resolved in https://github.com/alpenlabs/strata-p2p/pull/17/commits/0d807c64b90b3141078b328626387cd312c1d888